### PR TITLE
refactor(core): use utxolib for cashaddr

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
* Remove `bech32` format for bch (synonymous with `cashaddr`)
* Fix test vector that used mixed case instead of uppercase

BREAKING CHANGES:
* remove support for `bech32` parameter in (T)BCH `canonicalAddress()`

Issue: BG-40432

### CODE REVIEW

File: modules/utxo-lib/src/addressFormat.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  throw new Error(`could not parse outputScript [formats=${formats}]`);
+}
+
+/**
+ * Same as `toOutputScriptAndFormat`, only returning script
+ * @param address
+ * @param network
+ * @param formats
```
Comment (sangaman) [inline modules/utxo-lib/src/addressFormat.ts:100]: I think it's worth repeating the `defaults to all supported address formats for network` comment from above.

File: modules/utxo-lib/src/addressFormat.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+/**
+ * Attempts to parse address with different address formats, returns first hit.
+ * @param address
+ * @param network
+ * @param formats - defaults to all supported address formats for network
+ * @return tuple with [AddressFormat, Buffer] containing format and parsed output script
+ */
+export function toOutputScriptAndFormat(
```
Comment (sangaman) [inline modules/utxo-lib/src/addressFormat.ts:76]: It seems this function is only used in tests, are we ever going to need the format with the output script or can this be dropped?

**Review by sangaman — APPROVED** [review]

File: modules/utxo-lib/src/addressFormat.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/src/addressFormat.ts:76]: I think we might: we do have some code with format-specific handling. Not in utxolib directly though.

File: modules/utxo-lib/src/addressFormat.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/src/addressFormat.ts:100]: I'll submit a follow-up

**Pull request merged**